Provis.UnitTests/Core/Services/CommentServiceTests.cs

```diff
@@ -0,0 +1,271 @@
+using Provis.Core.Entities.WorkspaceTaskEntity;
+using Provis.Core.Exeptions;
+using Provis.Core.Interfaces.Repositories;
+using Provis.Core.Services;
+using Provis.UnitTests.Base.TestData;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Ardalis.Specification;
+using Provis.UnitTests.Base;
+using Provis.UnitTests.Resources;
+using System.Net;
+
+namespace Provis.UnitTests.Core.Services
+{
+    [TestFixture]
+    public class CommentServiceTests
+    {
+        protected CommentService _commentService;
+
+        protected Mock<UserManager<User>> _userManagerMock;
+        protected Mock<IRepository<User>> _userRepositoryMock;
+        protected Mock<IRepository<UserWorkspace>> _userWorkspaceRepositoryMock;
+        protected Mock<IRepository<Workspace>> _workspaceRepositoryMock;
+        protected Mock<IRepository<WorkspaceTask>> _taskRepositoryMock;
+        protected Mock<IRepository<Comment>> _commentRepositoryMock;
+        protected Mock<IMapper> _mapperMock;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _userManagerMock = UserManagerMock.GetUserManager<User>();
+            _workspaceRepositoryMock = new Mock<IRepository<Workspace>>();
+            _userWorkspaceRepositoryMock = new Mock<IRepository<UserWorkspace>>();
+            _userRepositoryMock = new Mock<IRepository<User>>();
+            _taskRepositoryMock = new Mock<IRepository<WorkspaceTask>>();
+            _commentRepositoryMock = new Mock<IRepository<Comment>>();
+            _mapperMock = new Mock<IMapper>();
+
+            _commentService = new CommentService(
+                _userRepositoryMock.Object,
+                _taskRepositoryMock.Object,
+                _commentRepositoryMock.Object,
+                _userWorkspaceRepositoryMock.Object,
+                _workspaceRepositoryMock.Object,
+                _userManagerMock.Object,
+                _mapperMock.Object);
+        }
+
+        [Test]
+        [TestCase("1")]
+        public async Task AddCommentAsync_ValidComment_ReturnCommentListDTO(string userId)
+        {
+            Comment comment = new()
+            {
+                DateOfCreate = new DateTimeOffset(DateTime.UtcNow, TimeSpan.Zero),
+                UserId = userId
+            };
+            CreateCommentDTO createCommentDTO = new()
+            {
+                CommentText = "Text",
+                TaskId = 1,
+                WorkspaceId = 1
+            };
+            
+            comment.CommentText = createCommentDTO.CommentText;
+            comment.TaskId = createCommentDTO.TaskId;
+
+            _mapperMock.Setup(x => 
+                x.Map(It.IsAny<CreateCommentDTO>(), It.IsAny<Comment>()))
+                    .Returns(comment);
+            SetupCommentAddAsync(comment);
+            SetupCommentSaveChangesAsync();
+
+            comment.Id = 1;
+            comment.User = new User() { UserName = "Username1" };
```

Comment:
Could you please make your code clean and move out duplication of object creation setups and ect